tests/test_extract_html_content.py

```diff
@@ -0,0 +1,313 @@
+
+            </activities>
+            </contents>
+        </moodle_backup>
+    """
+"""
+                <activity>
+                    <modulename>quiz</modulename>
+                    <directory>activities/quiz_3</directory>
+                </activity>
+                """
+
+
+def populate_tags(uuid_content1, uuid_content2, uuid_page):
+    PAGE2_CONTENT_TAG = f'<div class="os-raise-content"' \
+                        f' data-content-id="{uuid_page}"></div>'
+    LESSON1_CONTENT1_TAG = f'<div class="os-raise-content"' \
+                           f' data-content-id="{uuid_content1}"></div>'
+    LESSON1_CONTENT2_TAG = f'<div class="os-raise-content"' \
+                           f' data-content-id="{uuid_content2}"></div>'
+
+    LESSON1_CONTENT_TAGGED = f"""
+        <?xml version="1.0" encoding="UTF-8"?>
+<activity id="1" modulename="lesson">
+            <lesson id="1">
+                <name>First Lesson: 1.1</name>
+                <pages>
+                    <page id="3">
+                        <title>First Lession: 1.1 - lesson.xml</title>
+                        <contents>{(html.escape(LESSON1_CONTENT1_TAG,
+                                                quote=False))}</contents>
+                        <answers>
+                            <answer_text>{(html.escape(LESSON_ANSWER1,
+                                                       quote=False))}</answer_text>
+                            <answer_text>{(html.escape(LESSON_ANSWER2,
+                                                       quote=False))}</answer_text>
+                        </answers>
+                    </page>
+                    <page id="4">
+                        <title>Second Lession: 2.1 - lesson.xml</title>
+                        <contents>{(html.escape(LESSON1_CONTENT2_TAG,
+                                                quote=False))}</contents>
+                        <answers>
+                            <answer_text>{(html.escape(LESSON_ANSWER1,
+                                                       quote=False))}</answer_text>
+                            <answer_text>{(html.escape(LESSON_ANSWER2,
+                                                       quote=False))}</answer_text>
+                        </answers>
+                    </page>
+                </pages>
+            </lesson>
+        </activity>""".strip()
+    PAGE2_CONTENT_TAGGED = f"""
+        <?xml version="1.0" encoding="UTF-8"?>
+<activity id="2" modulename="page">
+            <page id="2">
+                <name>Some Moodle Hosted Content</name>
+                <content>{(html.escape(PAGE2_CONTENT_TAG,
+                                       quote=False))}</content>
+            </page>
+        </activity>
+    """.strip()
+    return [LESSON1_CONTENT_TAGGED, PAGE2_CONTENT_TAGGED]
+
+
+def replace_ampersand(content):
+    return content.replace("&amp;", "&").replace("&", "&amp;")
+
+
+@pytest.fixture
+def mbz_path(tmp_path):
```

Comment:
I think it would be better to get these two branches (109 and 91) merged on main and then think about how to re-use fixtures. The reason being that the two branches could share a fixture. We should have a separate issue for this.